console-subscriber/src/aggregator.rs

```diff
@@ -210,36 +342,89 @@ impl Aggregator {
+        let now = SystemTime::now();
+        let has_watchers = !self.watchers.is_empty();
+        drop_closed(
+            now,
+            &mut self.tasks,
+            &mut self.task_stats,
+            self.retention,
+            has_watchers,
+        );
+        drop_closed(
+            now,
+            &mut self.resources,
+            &mut self.resource_stats,
+            self.retention,
+            has_watchers,
+        );
+        drop_closed(
+            now,
+            &mut self.async_ops,
+            &mut self.async_op_stats,
+            self.retention,
+            has_watchers,
+        );
+    }
+
     /// Add the task subscription to the watchers after sending the first update
-    fn add_task_subscription(&mut self, subscription: Watch<proto::tasks::TaskUpdate>) {
-        tracing::debug!("new tasks subscription");
-        let new_tasks = self
-            .tasks
-            .all()
-            .map(|(id, task)| task.to_proto(id.clone()))
-            .collect();
+    fn add_instrument_subscription(
+        &mut self,
+        subscription: Watch<proto::instrument::InstrumentUpdate>,
+    ) {
+        tracing::debug!("new instrument subscription");
         let now = SystemTime::now();
-        let stats_update = self
-            .stats
-            .all()
-            .map(|(id, stats)| (id.into_u64(), stats.to_proto()))
-            .collect();
         // Send the initial state --- if this fails, the subscription is already dead
-        if subscription.update(&proto::tasks::TaskUpdate {
+        let update = &proto::instrument::InstrumentUpdate {
+            task_update: Some(proto::tasks::TaskUpdate {
+                new_tasks: self
+                    .tasks
+                    .as_proto(Include::All)
+                    .values()
+                    .cloned()
+                    .collect(),
+                stats_update: self.task_stats.as_proto(Include::All),
+            }),
+            resource_update: Some(proto::resources::ResourceUpdate {
+                new_resources: self
+                    .resources
+                    .as_proto(Include::All)
+                    .values()
+                    .cloned()
+                    .collect(),
+                stats_update: self.resource_stats.as_proto(Include::All),
+            }),
+            async_op_update: Some(proto::async_ops::AsyncOpUpdate {
+                new_async_ops: self
+                    .async_ops
+                    .as_proto(Include::All)
+                    .values()
+                    .cloned()
+                    .collect(),
+                stats_update: self.async_op_stats.as_proto(Include::All),
+            }),
+            resource_op_update: Some(proto::resource_ops::ResourceOpUpdate {
+                new_resource_ops: self
+                    .resource_ops
+                    .as_proto(Include::All)
+                    .values()
+                    .cloned()
+                    .collect(),
```

Comment:
`as_proto` returns a `HashMap`, which is produced by a call to `collect`. this means that in the case of the `new_xxxx` fields, where we don't want a `HashMap`, but a `Vec` of its values, we collect the iterator to a hashmap, iterate over the values, clone them, and then collect those values into a new `Vec`.

I think we should be able to avoid some unnecessary cloning/allocating if we just did this instead:
```suggestion
                new_tasks: self
                    .tasks
                    .all()
                    .map(|(_, value)| value.to_proto())
                    .collect()
                stats_update: self.task_stats.as_proto(Include::All),
            }),
            resource_update: Some(proto::resources::ResourceUpdate {
                new_resources: self
                    .resources
                    .all()
                    .map(|(_, value)| value.to_proto())
                    .collect()
                stats_update: self.resource_stats.as_proto(Include::All),
            }),
            async_op_update: Some(proto::async_ops::AsyncOpUpdate {
                new_async_ops: self
                    .async_ops
                    .all()
                    .map(|(_, value)| value.to_proto())
                    .collect()
                stats_update: self.async_op_stats.as_proto(Include::All),
            }),
            resource_op_update: Some(proto::resource_ops::ResourceOpUpdate {
                new_resource_ops: self
                    .resource_ops
                    .all()
                    .map(|(_, value)| value.to_proto())
                    .collect()
```
which should eliminate the unnecessary cloning/allocating